libs/platform/ebpf_epoch.c

```diff
@@ -714,19 +660,29 @@ _ebpf_epoch_hash_thread_id(uintptr_t thread_id)
     return (v2 >> 8) ^ (v2 & 0xFF);
 }
 
-_Requires_lock_held_(cpu_entry->lock) static ebpf_epoch_thread_entry_t* _ebpf_epoch_get_thread_entry(
-    _Inout_ ebpf_epoch_cpu_entry_t* cpu_entry, uintptr_t thread_id)
+_Requires_lock_held_(cpu_entry->lock) static ebpf_epoch_state_t* _ebpf_epoch_next_available_epoch_entry(
+    _Inout_ ebpf_epoch_cpu_entry_t* cpu_entry)
 {
+    uintptr_t thread_id = ebpf_get_current_thread_id();
     // Find the ideal bucket for this thread_id.
-    size_t bucket = _ebpf_epoch_hash_thread_id(ebpf_get_current_thread_id()) % EBPF_EPOCH_THREAD_ENTRY_TABLE_SIZE;
+    size_t bucket = _ebpf_epoch_hash_thread_id(thread_id) % EBPF_EPOCH_THREAD_TABLE_SIZE;
     // Search for the thread_id in the table starting at the ideal bucket.
-    for (size_t i = 0; i < EBPF_EPOCH_THREAD_ENTRY_TABLE_SIZE; i++) {
-        ebpf_epoch_thread_entry_t* thread_entry = &cpu_entry->thread_entry_table[bucket];
-        if (thread_entry->thread_id == thread_id) {
-            return thread_entry;
+    for (size_t i = 0; i < EBPF_EPOCH_THREAD_TABLE_SIZE; i++) {
+        ebpf_epoch_state_t* epoch_state = &cpu_entry->epoch_table[bucket];
+        if (epoch_state->epoch == 0) {
+            return epoch_state;
         }
-        bucket = (bucket + 1) % EBPF_EPOCH_THREAD_ENTRY_TABLE_SIZE;
+        bucket = (bucket + 1) % EBPF_EPOCH_THREAD_TABLE_SIZE;
     }
-    ebpf_assert(!"Thread entry not found");
+    ebpf_assert(!"Epoch state not found");
     return NULL;
 }
+
+uint32_t
+_ebpf_epoch_get_cpu_id_from_state(_In_ ebpf_epoch_state_t* state)
```

Comment:
Is there any reason to use the struct `ebpf_epoch_state_t` at all? Why can't we simply use `uint64_t` ?